packages/eas-cli/src/commands/worker/deploy.ts

```diff
@@ -0,0 +1,195 @@
+import fs from 'node:fs/promises';
+import * as path from 'node:path';
+
+import EasCommand from '../../commandUtils/EasCommand';
+import Log from '../../log';
+import { ora } from '../../ora';
+import { createProgressTracker } from '../../utils/progress';
+import * as WorkerAssets from '../../worker/assets';
+import { getSignedDeploymentUrlAsync } from '../../worker/deployment';
+import { UploadParams, batchUploadAsync, uploadAsync } from '../../worker/upload';
+
+const isDirectory = (directoryPath: string): Promise<boolean> =>
+  fs
+    .stat(directoryPath)
+    .then(stat => stat.isDirectory())
+    .catch(() => false);
+
+export default class WorkerDeploy extends EasCommand {
+  static override description = 'deploy an Expo web build';
+  static override aliases = ['deploy'];
+
+  // TODO(@kitten): Keep command hidden until worker deployments are live
+  static override hidden = true;
+  static override state = 'beta';
+
+  static override flags = {
+    // TODO(@kitten): Allow deployment identifier to be specified
+  };
+
+  static override contextDefinition = {
+    ...this.ContextOptions.ProjectConfig,
+    ...this.ContextOptions.ProjectDir,
+    ...this.ContextOptions.LoggedIn,
+  };
+
+  async runAsync(): Promise<void> {
+    Log.warn('EAS Worker Deployments are in beta and subject to breaking changes.');
+
+    const {
+      privateProjectConfig: { projectId, exp },
+      loggedIn: { graphqlClient },
+      projectDir,
+    } = await this.getContextAsync(WorkerDeploy, {
+      nonInteractive: true,
+    });
+
+    const distPath = path.resolve(projectDir, 'dist');
+    const distClientPath = path.resolve(distPath, 'client');
+    const distServerPath = path.resolve(distPath, 'server');
+    if (!(await isDirectory(distPath))) {
+      throw new Error(
+        `No "dist/" folder found at ${distPath}. Prepare your project for deployment with "npx expo export"`
+      );
+    } else if (!(await isDirectory(distClientPath))) {
+      throw new Error(
+        `No "dist/client/" folder found at ${distClientPath}. Ensure the app.json key "expo.web.output" is set to "server"`
+      );
+    } else if (!(await isDirectory(distServerPath))) {
+      throw new Error(
+        `No "dist/server/" folder found in ${distServerPath}. Ensure the app.json key "expo.web.output" is set to "server"`
+      );
+    }
+
+    async function* emitWorkerTarballAsync(
+      assetMap: WorkerAssets.AssetMap
+    ): AsyncGenerator<WorkerAssets.FileEntry> {
+      yield ['assets.json', JSON.stringify(assetMap)];
+
+      // TODO: Create manifest from user configuration
+      const manifest = { env: {} };
+      yield ['manifest.json', JSON.stringify(manifest)];
+
+      const workerFiles = WorkerAssets.listWorkerFiles(distServerPath);
+      for await (const workerFile of workerFiles) {
+        yield [`server/${workerFile.normalizedPath}`, workerFile.data];
+      }
+    }
+
+    async function uploadTarballAsync(tarPath: string): Promise<any> {
```

Comment:
I believe we expect `json.result` to have a certain structure later on. Can we validate that structure here using Joi/zod and return a typed value instead of `any`.

Reply:
We should eventually, but currently there is no result we use from the response body, and we do a simple `!json.success || !json.result || typeof json.result !== 'object'` sanity check. The response shape is currently only `success: boolean`, so there's no much data to convey just yet.

I'd like to keep this simple until we have a "real" payload to respond with, which is on the list of todos ✌️